specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/AppSpaceResources.json

```diff
@@ -0,0 +1,463 @@
+          "type": "string"
+        }
+      }
+    },
+    "RepositoryInformation": {
+      "description": "App Space Resource Repository Information - used to track the github repository for an App Space Resource's tracked ARM resource",
+      "type": "object",
+      "properties": {
+        "url": {
+          "description": "url for the repository",
+          "type": "string"
+        },
+        "branch": {
+          "description": "branch in the repository",
+          "type": "string"
+        }
+      }
+    },
+    "AppSpaceResourceProperties": {
+      "description": "AppSpaceResource resource-specific properties",
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "ARM resource Id of the tracked resource.",
+          "type": "string"
+        },
+        "friendlyName": {
+          "description": "Friendly name of the App Space Resource.",
+          "type": "string"
+        },
+        "resourceLocation": {
+          "description": "Location of the tracked ARM resource. Can be different from the location of the app space resource.",
+          "type": "string"
+        },
+        "repo": {
+          "$ref": "#/definitions/RepositoryInformation",
+          "description": "Optional App Space Resource Repository Information"
+        },
+        "deployment": {
+          "$ref": "#/definitions/DeploymentInformation",
+          "description": "Optional App Space Resource Deployment Information"
+        },
+        "isFavorite": {
+          "description": "Boolean marking whether or not this App Space Resource is a favorite (true or false, default is false)",
+          "type": "boolean"
+        },
+        "metadata": {
+          "description": "Property bag to add properties used by Portal UX.",
+          "type": "object"
+        },
+        "dependencies": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of ARM App Space Resource Ids on which this App Space Resource is dependent."
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "AppSpaceResource": {
+      "description": "App Space Resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AppSpaceResourceProperties"
+        }
+      }
+    },
+    "AppSpaceResourcePatch": {
+      "description": "AppSpaceResource patch object - Patch is done via a json merge and does not require all parameters for TrackedResource (i.e. location) to be present.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
```

Comment:
>properties [](http://example.com/codeflow?start=7&length=10)

If your resource requires tags and location, it should be a tracked resource, so you should use the TrackedResource definition from common-types (instead of Resource).

Reply:
I tried using TrackedResource but that was invalid for the Patch method because in TrackedResource, location is required. Do you have an alternate recommendation?
LintDiff failed when I had tracked resource here.

Reply:
Per conversation on teams, changing to not have "location" as one of the inputs for the PATCH methods - only having tags, common "Resource", and appspaceresourceproperties.